src/azure-cli/azure/cli/command_modules/acs/agentpool_decorator.py

```diff
@@ -180,20 +259,86 @@ def __init__(
         # store the context in the process of assemble the AgentPool object
         self.context = AKSAgentPoolContext(cmd, raw_parameters, self.models, decorator_mode=DecoratorMode.CREATE)
 
+    def _ensure_agentpool(self, agentpool: AgentPool) -> None:
+        """Internal function to ensure that the incoming `agentpool` object is valid and the same as the attached
+        `agentpool` object in the context.
+
+        If the incomding `agentpool` is not valid or is inconsistent with the `agentpool` in the context, raise a
```

Comment:
typo?

Reply:
incoming?

Reply:
oops

Reply:
Updated in commit [#1c010d9](https://github.com/Azure/azure-cli/pull/21543/commits/1c010d946e9d5172dcba35e94d08ddcdc8e34887).